app/src/main/java/org/sagebionetworks/research/mpower/tracking/TrackingMenuFragment.java

```diff
@@ -232,7 +231,12 @@ private void setIconListeners() {
 
                     passiveGaitViewModel.disableTracking();
 
-                    launcher.launchTask(getContext(), taskIdentifier, uuid, taskResult);
+                    Fragment parentFragment = this.getParentFragment();
+                    if (taskIdentifier.equals(WALK_AND_BALANCE) && TrackingTabFragment.class.equals(parentFragment.getClass())) {
```

Comment:
I would like to not create static variables inside of classes. But I see that it was before, I guess we can leave it